### PR TITLE
feat(db): implement ZDIFFSTORE command

### LINKED ISSUE #3887: implement ZDIFFSTORE
For reference, see https://valkey.io/commands/zdiffstore/

### PR DESCRIPTION
**Pull request by Elad-Mati**
Implement the ZDIFFSTORE command for sorted sets, which computes the difference between multiple sorted sets and stores the result in a new key.

Seems To be a not common bug in  ZDiffInternal() regarding  synchronization of fetching maps data from shards. Using instead   Execute(cb, true); caused deadlock.

Closes #3887

### CODE REVIEW

File: src/server/zset_family.cc [hunk 1]
```diff
@@ -2629,6 +2629,81 @@ void ZSetFamily::ZUnionStore(CmdArgList args, const CommandContext& cmd_cntx) {
   ZBooleanOperation(args, "zunionstore", true, true, cmd_cntx.tx, cmd_cntx.rb);
 }
 
+std::vector<ScoredMemberView> ZDiffInternal(CmdArgList args, const CommandContext& cmd_cntx,
+                                            size_t* maps_size_ptr = nullptr) {
+  vector<vector<ScoredMap>> maps(shard_set->size());
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    maps[shard->shard_id()] = OpFetch(shard, t);
```
Comment (adiholden) [inline src/server/zset_family.cc:2637]: do you think that opfetch in this flow will / will not fetch the destination key of the zdiffstore command?

Comment (adiholden) [inline src/server/zset_family.cc:2637]: do you think that opfetch in this flow will / will not fetch the destination key of the zdiffstore command?

File: src/server/zset_family.cc [hunk 2]
```diff
@@ -2629,6 +2629,81 @@ void ZSetFamily::ZUnionStore(CmdArgList args, const CommandContext& cmd_cntx) {
...
+    return OpStatus::OK;
+  };
+  // cmd_cntx.tx->Execute(cb, false);
+  cmd_cntx.tx->ScheduleSingleHop(std::move(cb));
```
Comment (adiholden) [inline src/server/zset_family.cc:2641]: you should use the Execute api for multi hops commands

File: src/server/zset_family.cc [hunk 3]
```diff
@@ -2629,6 +2629,81 @@ void ZSetFamily::ZUnionStore(CmdArgList args, const CommandContext& cmd_cntx) {
...
+  if (nullptr != maps_size_ptr) {
+    *maps_size_ptr = maps.size();
+  }
+  // Extract the ScoredMap of the first key
+  auto& sm = maps[sid];
+  if (sm.empty()) {
+    return {};
+  }
+
+  auto result = std::move(sm[0]);
```
Comment (adiholden) [inline src/server/zset_family.cc:2655]: The code bellow is similar to the one in zdiff command therefore it should be extracted to a function

File: src/server/zset_family.cc [hunk 4]
```diff
@@ -2629,6 +2629,81 @@ void ZSetFamily::ZUnionStore(CmdArgList args, const CommandContext& cmd_cntx) {
...
+  size_t maps_size = 0;
+  auto smvec = ZDiffInternal(modifiedArgs, cmd_cntx, &maps_size);
+
+  if (smvec.empty()) {
+    cmd_cntx.rb->SendLong(0);
+    return;
+  }
+
+  string_view dest_key = ArgS(args, 0);
+  auto store_cb = [&, dest_shard = Shard(dest_key, maps_size)](Transaction* t, EngineShard* shard) {
```
Comment (adiholden) [inline src/server/zset_family.cc:2696]: the shard_id calculation should be based on the shard_set->size and not map_size

File: src/server/zset_family_test.cc [hunk 5]
```diff
@@ -1123,4 +1123,40 @@ TEST_F(ZSetFamilyTest, RangeStore) {
   EXPECT_THAT(resp, ArrLen(0));
 }
 
+TEST_F(ZSetFamilyTest, ZDiffStore) {
+  RespExpr resp;
+
+  resp = Run({"zdiffstore", "key", "0"});
```
Comment (adiholden) [inline src/server/zset_family_test.cc:1129]: I suggest to check more edge cases for invalid command
you can take an example from ZDiffError test
